### PR TITLE
[ListVnext] Added bolded header style

### PR DESCRIPTION
**Pull request by sophialee0416**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

A continuation of #424 :
- Added new Header style. This style is larger and bolded, and is the vNext parity of current FluentUI's HeaderFooter Primary style.
- Reorganized dividers logic between sections + cells

### Verification

| Before                                      |  After                                    |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 11 Pro - 2021-02-23 at 16 35 17 | Simulator Screen Shot - iPhone 11 Pro - 2021-02-23 at 16 34 01 |

The header theming test below ensures that the tokens used in the Header are able to overridden as determined by the client app. These will not be merged with the PR.

https://user-images.githubusercontent.com/22566866/109076855-df795000-76c0-11eb-98fa-5f6987eef46b.mov

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/447)

### CODE REVIEW

**Review by rdeassis — COMMENTED** [review]
I took the liberty of editing the PR description to fix the table with "Before" and "After" headers.
The videos unfortunately don't play well inside tables. So, I moved it down below the table.

A few asks:
1. Can you upload the video again so we can get the preview on the PR description?
2. Can you also provide a description of the theming validation before the video? (explaining what you have changed in the code, what you intended to validate and state that those changes are not being merged, but only for theming validation purposes)
3. Validate the podspec by running "pod lib lint MicrosoftFluentUIVnext.podspec" and ensuring you get a passing validation

File: ios/FluentUI/Vnext/Core/FluentUIStyle.generated.swift [hunk 1]
```diff
@@ -2296,4 +2301,4 @@ open class FluentUIStyle: NSObject {
 		}
 	}
 
-}
+}
```
[the remark this answers is not in the document]
Comment (sophialee0416) [reply ios/FluentUI/Vnext/Core/FluentUIStyle.generated.swift:2304]: Didn't touch it, that's weird. Should be fixed with the changes

File: ios/FluentUI/Vnext/List/ListTokens.swift [hunk 2]
```diff
@@ -61,15 +56,10 @@ public class MSFListTokens: MSFTokensBase, ObservableObject {
         let currentTheme = theme
         let appearanceProxy = currentTheme.MSFListTokens
 
-        backgroundColor = appearanceProxy.backgroundColor
+        backgroundColor = appearanceProxy.backgroundColor.rest
```
[the remark this answers is not in the document]
Comment (sophialee0416) [reply ios/FluentUI/Vnext/List/ListTokens.swift]: Removed backgroundColor entirely as it is now being handled by ListCell only

File: tools/sgen/input/FluentUIStyle.yml [hunk 3]
```diff
@@ -420,17 +421,11 @@ MSFGhostButtonTokens extends MSFButtonTokens:
...
-  backgroundColor: $Colors.Background.neutral1
   borderColor: $Colors.Stroke.neutral1
-  borderSize: $Border.size.thin
-  horizontalCellPadding: $Spacing.medium
-  sublabelColor: $Colors.Foreground.neutral3
-  sublabelFont: $Typography.caption1
 
 AP_MSFListCellTokens:
   backgroundColor: [ rest: $Colors.Background.neutral1, pressed: $Colors.Background.neutral5 ]
   borderColor: $Colors.Stroke.neutral1
```
Comment (rdeassis) [inline tools/sgen/input/FluentUIStyle.yml:428]: This borderColor is not being used.
Is this the color that gets applied to the cell divider? (ListCell.swift, line 114)
If this is indeed not used, let's delete it from here, generated files and the ListTokens.swift (line 62)!

Comment (rdeassis) [reply tools/sgen/input/FluentUIStyle.yml:428]: Thanks for the offline conversation on the "Divider x Rectangle" issue! 

I think we're good to go with one of them based on what the design team decides with regards to the thickness (Rectangle if they want full control of the thickness versus Divider if they just want to have the default value from the platform).

I believe this is the last standing issue on the PR before we ca merge it.

**Pull request merged**